### PR TITLE
liana: remove daemon feature

### PR DESCRIPTION
**Pull request by edouardparis**
We expect user to use systemd service to manage a daemon or
to handle their process by themself.

The json-rpc server is exposed only for unix systems through
unix sockets, it is moved to its own module while the implementation
for windows is marked as TODO.

### CODE REVIEW

**Comment by pythcoiner** [comment]
she should check if #255 and #414 can be closed after merging this (but i don't believe libc was used only for forking the process...)

**Comment by edouardparis** [comment]
You are right, libc is required by rust anyway to call OS-level functions and system calls directly. I do not think the current change resolves #255 and #414

File: liana/src/config.rs [hunk 1]
```diff
@@ -322,7 +312,6 @@ mod tests {
         {
             let toml_str = r#"
             data_dir = '/home/wizardsardine/custom/folder/'
-            daemon = false
```
Comment (jp1ac4) [inline liana/src/config.rs:325]: I think you could now remove the `#[cfg(unix)]` attribute above and elsewhere in this test.

**Review by jp1ac4 — COMMENTED** [review]
I was able to run lianad, executing CLI commands and running the GUI.

As well as my comment in the code, I think you can also remove the `StartupError::Daemonization` variant now.

**Review by jp1ac4 — APPROVED** [review]
ACK https://github.com/wizardsardine/liana/commit/c14c55beeea56655096eb873ba0bce4d37ccd6bb.

**Comment by pythcoiner** [comment]
tACK c14c55b run GUI and lianad

**Pull request merged**